setup.py

```diff
@@ -42,13 +42,15 @@
             "astroquery",
             "scipy",
             "rdflib",
-            "black"
+            "black",
+            "bokeh"
```

Comment:
In dispatcher, we fixed the bokeh version to be exactly `2.4.2`
The purpose of this is that it has to coincide with the version of the accompanying js library in the frontend. 

On the other hand, it's probably a good time to update it to the latest version, which is currently 3.5.1. One of the motivations is that, strictly speaking, the support for python 3.10 was added in bokeh 3.0, and we test oda_api package against 3.8 - 3.11

Reply:
Some complication is that it should be done in coordinated fashion between oda_api, dispatcher, frontend, and possibly gallery (not sure, please confirm)

Reply:
> same for the frontend

Frontend serves bokeh js libs that are used in conjunction with what dispatcher returns. That's why I also thought about gallery, if there are any bokeh-based products there. 
So the change is small, just to updating the version, but it's backwards-incompatible.

Reply:
> We can probably drop 3.8 and maybe 3.9

Well, 3.9 still has a year of lifecycle, but [Bokeh is officially supported and tested on Python 3.10 and above (CPython).](https://docs.bokeh.org/en/latest/docs/first_steps.html)
Still, I suspect it will work with it

Reply:
oda-api should run in different environments, sometimes pretty old ones. For legacy purposes. So we should keep as dependency as broad as possible.

Reply:
with https://github.com/oda-hub/oda_api/pull/279/commits/693e88deb99149dbe4f32dfd9baf204f9e244fb6 the html output for LCs will have bokeh-3.5.1

Reply:
> with [693e88d](https://github.com/oda-hub/oda_api/commit/693e88deb99149dbe4f32dfd9baf204f9e244fb6) the html output for LCs will have bokeh-3.5.1

Need to try, how it works with frontend, because it injects bokeh==2.4.2 in a page head

Reply:
> with [693e88d](https://github.com/oda-hub/oda_api/commit/693e88deb99149dbe4f32dfd9baf204f9e244fb6) the html output for LCs will have bokeh-3.5.1

I forgot to specify, the update is about the `get_html_image` function used to generate LC plots for the gallery.

For the frontend, a dedicated PR is needed on the dispatcher

Reply:
I didn't quite follow how it works in gallery. Each plot there have their own bokeh version injected, and they don't conflict because each product is on a different page, right? 
In frontend, it may be more complicated, say some products are cached with previous version of bokeh, and some other is generated with recent version injected. But all the tabs are on the same page, and here we may encounter problems

Reply:
> I didn't quite follow how it works in gallery. Each plot there have their own bokeh version injected, and they don't conflict because each product is on a different page, right? 

Exactly, more specifically, it is done here:
https://github.com/oda-hub/oda_api/blob/02e47e3670542e763f381364c4a87de99a2a9de9/oda_api/plot_tools.py#L715-L719

> In frontend, it may be more complicated, say some products are cached with previous version of bokeh, and some other is generated with recent version injected. But all the tabs are on the same page, and here we may encounter problems

Yes I can totally see it. And a potential workaround would require probably a lot of work. is it for sure not retro-compatible ?

Reply:
I saw failures when bokeh version in python and js are different. Between major releases, they surely won't be compatible.